shanoir-ng-front/src/app/home/home.component.ts

```diff
@@ -30,11 +31,49 @@ export class HomeComponent {
 
     shanoirBigLogoUrl: string = ImagesUrlUtil.SHANOIR_BLACK_LOGO_PATH;
     
-    private http: HttpClient = ServiceLocator.injector.get(HttpClient);
+    challengeDua: DataUserAgreement;
+    challengeStudy: Study;
 
-    constructor(private breadcrumbsService: BreadcrumbsService) {
+    constructor(
+            private breadcrumbsService: BreadcrumbsService,
+            private studyService: StudyService) {
         //this.breadcrumbsService.nameStep('Home');
         this.breadcrumbsService.markMilestone();
+        this.studyService.getMyDUA().then(duas => {
+            if (duas) {
+                for (let dua of duas) {
+                    if (dua.isChallenge) {
+                        this.challengeDua = dua;
+                        return;
+                    }
+                }
+            }
+        }).then(() => {
+            if (!this.challengeDua) {
+                this.fetchChallengeStudy()
+            }
+        });
+    }
+
+    onSign() {
+        this.fetchChallengeStudy();
+    }
+
+    private fetchChallengeStudy() {
+        this.studyService.getAll().then(studies => {
+            if (studies) {
+                for (let study of studies) {
+                    if (study.challenge) {

```

Comment:
So we hope not to get multiple challenges here ?
It's OK for me too, as normally a user is subscribing to shanoir only for the challenge

Reply:
You are right, this version supports only one challenge per user